postgres-protocol/src/misc/mod.rs

```diff
@@ -0,0 +1,92 @@
+//! As always, there are some functions that just don't fit anywhere.
+
+use hmac::{Hmac, Mac, NewMac};
+use rand::RngCore;
+use sha2::digest::FixedOutput;
+use sha2::{Digest, Sha256};
+
+#[cfg(test)]
+mod test;
+
+const SCRAM_DEFAULT_ITERATIONS: u32 = 4096;
+const SCRAM_DEFAULT_SALT_LEN: usize = 16;
+
+/// Choose the password encryption algorithm.
+#[derive(Debug)]
+pub enum PasswordAlgorithm {
```

Comment:
This should be `#[non_exhaustive]` so we can add more variants in the future.

Reply:
Resolved by reworking the API.